disnake/automod.py

```diff
@@ -0,0 +1,681 @@
+        else:
+            presets = None
+
+        return cls(  # type: ignore  # call doesn't match any overloads
+            keyword_filter=data.get("keyword_filter"),
+            presets=presets,
+            allow_list=data.get("allow_list"),
+        )
+
+    def to_dict(self) -> AutoModTriggerMetadataPayload:
+        data: AutoModTriggerMetadataPayload = {}
+        if self.keyword_filter is not None:
+            data["keyword_filter"] = list(self.keyword_filter)
+        if self.presets is not None:
+            data["presets"] = self.presets.values  # type: ignore  # `values` contains ints instead of preset literal values
+        if self.allow_list is not None:
+            data["allow_list"] = list(self.allow_list)
+        return data
+
+    def __repr__(self) -> str:
+        s = f"<{type(self).__name__}"
+        if self.keyword_filter is not None:
+            s += f" keyword_filter={self.keyword_filter!r}"
+        if self.presets is not None:
+            s += f" presets={self.presets!r}"
+        if self.allow_list is not None:
+            s += f" allow_list={self.allow_list!r}"
+        return f"{s}>"
+
+
+class AutoModRule:
+    """
+    Represents an auto moderation rule.
+
+    .. versionadded:: 2.6
+
+    Attributes
+    ----------
+    id: :class:`int`
+        The rule ID.
+    name: :class:`str`
+        The rule name.
+    enabled: :class:`bool`
+        Whether this rule is enabled.
+    guild: :class:`Guild`
+        The guild of the rule.
+    creator_id: :class:`int`
+        The rule creator's ID. See also :attr:`.creator`.
+    event_type: :class:`AutoModEventType`
+        The event type this rule is applied to.
+    trigger_type: :class:`AutoModTriggerType`
+        The type of trigger that determines whether this rule's actions should run for a specific event.
+    trigger_metadata: :class:`AutoModTriggerMetadata`
+        Additional metadata associated with this rule's :attr:`.trigger_type`.
+    exempt_role_ids: FrozenSet[:class:`int`]
+        The role IDs that are exempt from this rule.
+    exempt_channel_ids: FrozenSet[:class:`int`]
+        The channel IDs that are exempt from this rule.
+    """
+
+    __slots__ = (
+        "id",
+        "name",
+        "enabled",
+        "guild",
+        "creator_id",
+        "event_type",
+        "trigger_type",
+        "trigger_metadata",
+        "_actions",
+        "exempt_role_ids",
+        "exempt_channel_ids",
+    )
+
+    def __init__(self, *, data: AutoModRulePayload, guild: Guild):
+        self.guild: Guild = guild
+
+        self.id: int = int(data["id"])
+        self.name: str = data["name"]
+        self.enabled: bool = data["enabled"]
+        self.creator_id: int = int(data["creator_id"])
+        self.event_type: AutoModEventType = try_enum(AutoModEventType, data["event_type"])
+        self.trigger_type: AutoModTriggerType = try_enum(AutoModTriggerType, data["trigger_type"])
```

Comment:
Thoughts on using `Final[AutoModTriggerType]`?

Although I suppose that could wait until a second pr to add Final typings to multiple attributes across the library which should not be changed (most ID attributes, specifically).

I still think it might be useful to add here.

Reply:
None of these attributes should be modified directly either way, there's this warning in the section that `AutoModRule` is in:
![image](https://user-images.githubusercontent.com/8530778/180662675-b1a1d4b7-5862-4b86-b11a-55ad0a954f0a.png)

Adding `Final` in more places in a future PR sounds fine though.